core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -519,6 +521,20 @@ private[spark] class MesosClusterScheduler(
     val currentTime = new Date()
 
     stateLock.synchronized {
+      var it = offers.iterator()
+      while (it.hasNext) {
+        val offer = it.next()
+        val offerAttributes = toAttributeMap(offer.getAttributesList)
+        val meetsConstraints = matchesAttributeRequirements(driverOfferConstraints, offerAttributes)
+
+        if (!meetsConstraints) {
+          driver.declineOffer(offer.getId)
+          it.remove()
```

Comment:
oh, it's true. I modify code base in 1.5.2 previously, and test ok in practice. in 1.6.0 version the code is change. I will revise it soon.thanks for your review.